src/FeedsCleanup/Program.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                return 3;
+            }
+
+            return 0;
+        }
+
+        private static bool RemoveFromCI(IEnumerable<string> coherentPackageNames, string ciLatestPackagesFolder, bool whatif)
+        {
+            var latestPackageFolders = Directory.GetDirectories(ciLatestPackagesFolder);
+
+            var deleteCandiates = new List<string>();
+
+            foreach (var latestPackageFolder in latestPackageFolders)
+            {
+                var projectName = Path.GetFileName(latestPackageFolder);
+
+                var matchedVersion = new HashSet<string>(
+                    coherentPackageNames
+                        .Where(packageName => packageName.IndexOf(projectName) == 0)
+                        .Select(packageName => packageName.Substring(projectName.Length + 1)));
+
+                if (matchedVersion.Any())
+                {
+                    deleteCandiates.AddRange(Directory.GetDirectories(latestPackageFolder)
+                                             .Where(p => !matchedVersion.Contains(Path.GetFileName(p))));
+                }
+                else
+                {
+                    deleteCandiates.Add(latestPackageFolder);
+                    Warn("Project {0} doesn't exist in this Coherence build hence it will be deleted from {1}", projectName, ciLatestPackagesFolder);
+                }
+            }
+
+            foreach (var candiate in deleteCandiates)
+            {
+                Info("Delete folder {0}", candiate);
+
+                if (!whatif)
+                {
+                    Info("Actual delete!");
+                    Directory.Delete(candiate, recursive: true);
+                }
+            }
+
+            return true;
+        }
+
+        private static bool RemoveFromdMyGet(IEnumerable<string> coherentPackageNames,
+                                             string feed,
+                                             string apiKey,
+                                             bool whatif)
+        {
+            Console.WriteLine("Listing packages on {0}", feed);
+            var task = Task.Run(() =>
+            {
+                var repo = PackageRepositoryFactory.Default.CreateRepository(feed);
+                return repo.GetPackages().ToList();
+            });
+
+            var remotePackagesIds = task.Result.Select(package => package.Id).Distinct();
+            var localPackagsFullNames = coherentPackageNames.Select(n => Path.GetFileNameWithoutExtension(n));
+
+            var deleteCandidateIds = new HashSet<string>();
+            foreach (var id in remotePackagesIds)
+            {
+                if (!localPackagsFullNames.Any(fullname => fullname.Contains(id)))
+                {
+                    deleteCandidateIds.Add(id);
+                }
+            }
+
+            var requests = task.Result.Where(package => deleteCandidateIds.Contains(package.Id))
```

Comment:
I don't think so, that's the reason we need to pull every version. But I can try.